dashboard/modules/serve/serve_agent.py

```diff
@@ -181,14 +181,15 @@ async def put_all_deployments(self, req: Request) -> Response:
                 text=error_msg,
             )
 
-        client = serve_start(
-            detached=True,
-            http_options={
-                "host": config.host,
-                "port": config.port,
-                "location": "EveryNode",
-            },
-        )
+        async with self._controller_lock:
```

Comment:
is the lock necessary? ray will handle the race condition with two processes trying to start the actor at the same time, so only one of these will go through

even with the lock we'll still need to handle the error case where two of these calls run sequentially

Reply:
Actually, I suppose it's slightly better because `serve start` has some handling to check if the actor exists already. Makes sense.

Can you please add a comment here describing why we have it and what the expected behavior is if two calls reach the lock line concurrently?

Reply:
Added comments. 
IMO Adding lock here makes our life easier to maintain the serve start logics, besides controller actor starts, we also have some logics `get_global_client` (involving blocking call with controller) and  `_set_global_client`, so it would be convenient for the caller to maintain a lock to get the serve instance started correctly :)